docs/manual/creating-a-checker.tex

```diff
@@ -569,25 +569,73 @@
 \item \refqualclass{framework/qual}{PolymorphicQualifier} denotes that a qualifier is a
   polymorphic qualifier.  For example:
 
-  \begin{Verbatim}
-    @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
-    @PolymorphicQualifier
-    public @interface PolyNull {}
-  \end{Verbatim}
+\begin{Verbatim}
+  import java.lang.annotation.ElementType;
+  import java.lang.annotation.Target;
+  import org.checkerframework.framework.qual.PolymorphicQualifier;
+
+  @PolymorphicQualifier
+  @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
+  public @interface PolyNull {}
+\end{Verbatim}
 
-  For a description of polymorphic qualifiers, see
-  Section~\ref{method-qualifier-polymorphism}.  A polymorphic qualifier must not have
+  A polymorphic qualifier must not have
   a \refqualclass{framework/qual}{SubtypeOf} meta-annotation nor be
   mentioned in any other \refqualclass{framework/qual}{SubtypeOf}
   meta-annotation.
 
+  For how to restrict instantiation, see Section~\ref{qualifier-polymorphism-define-restrict}.
+  For a general description of polymorphic qualifiers, see
+  Section~\ref{method-qualifier-polymorphism}.
+
 \end{itemize}
 
 The declarative and procedural mechanisms for specifying the hierarchy can
 be used together. If any of the annotations representing type qualifiers have elements, then
 the relationships between those qualifiers must be defined procedurally.
 
 
+\subsubsectionAndLabel{Restricting instantiation}{qualifier-polymorphism-define-restrict}
+
+Sometimes, you wish to restrict how a polymorphic qualifier can be
+instantiated.  An example is the Signedness Checker
+(\chapterpageref{signedness-checker}).  Its two main qualifiers are
+\refqualclass{checker/signedness/qual}{Signed} and
+\refqualclass{checker/signedness/qual}{Unsigned}, and the top qualifier
+(which should be rarely used) is
+\refqualclass{checker/signedness/qual}{UnknownSignedness}.
+
+As described so far, there is no way for a programmer to require that two
+method arguments have the same signedness type.  Writing
+
+\begin{Verbatim}
+  void foo(@PolySigned Object a, @PolySigned Object b) { ... }
+\end{Verbatim}
+
+\noindent
+permits every invocation, because one if its instantiations is
+
+\begin{Verbatim}
+  void foo(@UnknownSignedness Object a, @UnknownSignedness Object b) { ... }
+\end{Verbatim}
+
+\noindent
+which is consistent with any two arguments, regardless of their signedness
+type.
+
+It is possible to permit only method calls whose arguments have exactly the
```

Comment:
Should we say something about how this still allows upcasts to top?  Also if a type system designer is considering this, then the type system may need [class qualifier polymorphism](https://checkerframework.org/manual/#class-qualifier-polymorphism).